### PR TITLE
[Lib] Added skill switch dialog

### PR DESCRIPTION
**Pull request by lauren-mills**
### Purpose
*What is the context of this pull request? Why is it being done?*
Added skill switch dialog and options to lib

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

#### Bots
- [ ] I have validated that new and updated responses use appropriate [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) and [InputHint](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) properties to ensure a high-quality speech-first experience
- [ ] I have replicated language model changes across the English, French, Italian, German, Spanish, and Chinese `.lu` files and validated that deployment is successful

#### Deployment Scripts
- [ ] I have replicated my changes in the **Virtual Assistant Template** and **Sample** projects
- [ ] I have replicated my changes in the **Skill Template** and **Sample** projects

### CODE REVIEW

**Review by lzc850612 — CHANGES_REQUESTED** [review]
My feeling is that it's better to put them under the Skills folder. under Skills/Dialogs

File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Skills/Dialogs/SkillSwitchDialog.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            _lastActivityAccessor = conversationState.CreateProperty<Activity>(Properties.LastActivity);
+
+            var intentSwitch = new WaterfallStep[]
+            {
+                PromptToSwitchAsync,
+                EndAsync,
+            };
+
+            AddDialog(new WaterfallDialog(nameof(SkillSwitchDialog), intentSwitch));
+            AddDialog(new ConfirmPrompt("ConfirmIntentSwitch"));
```
Comment (lzc850612) [inline sdk/csharp/libraries/microsoft.bot.builder.solutions/Skills/Dialogs/SkillSwitchDialog.cs]: define a string for "ConfirmIntentSwitch" for reusing and you're good to go :)

**Review by lzc850612 — APPROVED** [review]

**Review by ryanisgrig — APPROVED** [review]
This is incredibly pedantic but I think it'd be better named as SwitchSkillDialog (verb-noun) than SkillSwitchDialog (noun-verb)

File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Skills/Dialogs/SwitchSkillDialogOptions.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SwitchSkillDialogOptions"/> class.
+        /// </summary>
+        /// <param name="prompt">The <see cref="Activity"/> to display when prompting to switch skills.</param>
+        /// <param name="manifest">The <see cref="SkillManifest"/> for the new skill.</param>
+        public SwitchSkillDialogOptions(Activity prompt, SkillManifest manifest)
+        {
+            Prompt = prompt;
+            Skill = manifest;
```
Comment (xieofxie) [inline sdk/csharp/libraries/microsoft.bot.builder.solutions/Skills/Dialogs/SwitchSkillDialogOptions.cs:16]: How about inherit this from PromptOptions? So ConfirmPrompt is fully configurable.

**Pull request merged**